Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -461,6 +468,49 @@ def add_comment(self, issue, text):
 
         return result
 
+    def relate(self, issue, depends_on=None, blocks=None, regressed_by=None, regressions=None, **relations):
+        if relations:
+            raise TypeError("'{}' is an invalid relation".format(list(relations.keys())[0]))
+
+        update_dict = dict()
+        update_dict['ids'] = [issue.id]
+        if depends_on:
+            update_dict['depends_on'] = {'add': depends_on}
+        if blocks:
+            update_dict['blocks'] = {'add': blocks}
+        if regressed_by:
+            update_dict['regressed_by'] = {'add': regressed_by}
+        if regressions:
+            update_dict['regressions'] = {'add': regressions}
+
+        response = None
+        try:
+            response = self.session.put(
+                '{}/rest/bug/{}{}'.format(self.url, issue.id, self._login_arguments(required=True)),
+                json=update_dict,
+                timeout=self.timeout,
+            )
+        except RuntimeError as e:
+            sys.stderr.write('{}\n'.format(e))
+        if response and response.status_code // 100 == 4 and self._logins_left:
+            self._logins_left -= 1
+        if not response or response.status_code // 100 != 2:
+            sys.stderr.write("Failed to modify '{}'\n".format(issue))
+            return None
+
+        if not issue._related:
```

Comment:
Should probably have an `else` here, I think. No need to explicitly update `issue._related` if we had to populate it, right?

Reply:
Good catch